cockatrice/src/client/ui/picture_loader/picture_loader_local.cpp

```diff
@@ -59,84 +59,81 @@ QImage PictureLoaderLocal::tryLoad(const CardInfoPtr &toLoad) const
+
+    if (setInstance.getPtr()) {
+        setName = setInstance.getPtr()->getCorrectedShortName();
+        collectorNumber = setInstance.getProperty("num");
+        providerId = setInstance.getProperty("uuid");
     }
 
     qCDebug(PictureLoaderLocalLog).nospace()
-        << "[card: " << cardName << " set: " << setName << "]: Skipping load picture from local";
-
-    return QImage();
+        << "[card: " << cardName << " set: " << setName << "]: Attempting to load picture from local";
+    return tryLoadCardImageFromDisk(setName, correctedCardName, collectorNumber, providerId);
 }
 
 QImage PictureLoaderLocal::tryLoadCardImageFromDisk(const QString &setName,
                                                     const QString &correctedCardName,
-                                                    const bool searchCustomPics) const
+                                                    const QString &collectorNumber,
+                                                    const QString &providerId) const
 {
     QImage image;
     QImageReader imgReader;
     imgReader.setDecideFormatFromContent(true);
-    QList<QString> picsPaths = QList<QString>();
-
-    if (searchCustomPics) {
-        picsPaths << customFolderIndex.values(correctedCardName);
-    }
 
-    if (!setName.isEmpty()) {
-        picsPaths << picsPath + "/" + setName + "/" + correctedCardName
-                  // We no longer store downloaded images there, but don't just ignore
-                  // stuff that old versions have put there.
-                  << picsPath + "/downloadedPics/" + setName + "/" + correctedCardName;
-    }
-
-    // Iterates through the list of paths, searching for images with the desired
-    // name with any QImageReader-supported extension
-    for (const auto &_picsPath : picsPaths) {
-        if (!QFileInfo(_picsPath).isFile()) {
+    // Order is cardName_providerId, cardName_setName_collectorNumber, setName-collectorNumber-cardName and then just
+    // generically cardName, if the user has decided to override every printing. Most-to-least specific.
+
+    const QStringList nameVariants = {!providerId.isEmpty() ? QString("%1_%2").arg(correctedCardName, providerId)
+                                                            : QString(),
+                                      (!setName.isEmpty() && !collectorNumber.isEmpty())
+                                          ? QString("%1_%2_%3").arg(correctedCardName, setName, collectorNumber)
+                                          : QString(),
+                                      (!setName.isEmpty() && !collectorNumber.isEmpty())
+                                          ? QString("%1-%2-%3").arg(setName, collectorNumber, correctedCardName)
+                                          : QString(),
+                                      correctedCardName};
+
+    for (const QString &nameVariant : nameVariants) {
+        if (nameVariant.isEmpty()) {
             continue;
         }
 
```

Comment:
Inlining the `nameVariants` string creation makes it kinda hard to follow, and it also adds empty string to the list that need to be skipped. I wonder if it would be nicer if we constructed the `nameVariants` list piece-by-piece.
```suggestion
    QStringList nameVariants;
    if (!providerId.isEmpty()) {
        nameVariants << QString("%1_%2").arg(correctedCardName, providerId);
    }
    if (!setName.isEmpty() && !collectorNumber.isEmpty()) {
        nameVariants << QString("%1_%2_%3").arg(correctedCardName, setName, collectorNumber)
                     << QString("%1-%2-%3").arg(setName, collectorNumber, correctedCardName);
    }
    nameVariants << correctedCardName;

    for (const QString &nameVariant : nameVariants) {```

Reply:
Yeah, that's reasonable as well.